core/src/main/java/org/jruby/internal/runtime/AbstractIRMethod.java

```diff
@@ -42,21 +44,40 @@ public AbstractIRMethod(IRScope method, Visibility visibility, RubyModule implem
         this.staticScope.determineModule();
         this.signature = staticScope.getSignature();
 
-        // -1 jit.threshold is way of having interpreter not promote full builds.
-        if (Options.JIT_THRESHOLD.load() == -1) callCount = -1;
+        final Ruby runtime = implementationClass.getRuntime();
+        if (runtime.getInstanceConfig().isJitDisabled()) setCallCount(-1);
```

Comment:
Would this be better as `! isJitEnabled()` and just reverse the logic? Something about `isSomethingNegative` methods bothers me.

Reply:
+1 will update